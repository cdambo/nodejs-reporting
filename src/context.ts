export default class Context {
  private readonly globalTags: object;

  public constructor(globalTags: object = {}) {
    this.globalTags = globalTags;
  }

  public mergeTags(tags: object = {}): object {
    return Object.assign({}, this.globalTags, tags);
  }
}
