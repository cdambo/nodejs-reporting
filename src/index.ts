import Context from "./context";

export * from "./metrics";
export { Context };
