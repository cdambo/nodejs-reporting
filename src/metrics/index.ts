export * from "./formats";
export * from "./reporters";
export * from "./integrations";
