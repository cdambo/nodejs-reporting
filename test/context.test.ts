import { Context } from "../src";

describe("Context", (): void => {
  describe("mergeTags", (): void => {
    it("Returns a merged object of tags", (): void => {
      const context = new Context({ a: 1 });
      expect(context.mergeTags({ b: 2 })).toMatchSnapshot();
    });

    describe("Empty globalTags", (): void => {
      it("Returns a merged object of tags", (): void => {
        const context = new Context();
        expect(context.mergeTags({ b: 2 })).toMatchSnapshot();
      });
    });

    describe("Empty tags", (): void => {
      it("Returns a merged object of tags", (): void => {
        const context = new Context({ a: 1 });
        expect(context.mergeTags()).toMatchSnapshot();
      });
    });

    describe("Undefined tags", (): void => {
      it("Returns a merged object of tags", (): void => {
        const context = new Context({ a: 1 });
        expect(context.mergeTags(undefined)).toMatchSnapshot();
      });
    });
  });
});
