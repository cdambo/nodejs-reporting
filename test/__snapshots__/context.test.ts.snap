// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context mergeTags Empty globalTags Returns a merged object of tags 1`] = `
Object {
  "b": 2,
}
`;

exports[`Context mergeTags Empty tags Returns a merged object of tags 1`] = `
Object {
  "a": 1,
}
`;

exports[`Context mergeTags Returns a merged object of tags 1`] = `
Object {
  "a": 1,
  "b": 2,
}
`;

exports[`Context mergeTags Undefined tags Returns a merged object of tags 1`] = `
Object {
  "a": 1,
}
`;
